tfhe/src/integer/server_key/radix/mod.rs

```diff
@@ -756,4 +757,48 @@ impl ServerKey {
             self.propagate(ctxt, i);
         }
     }
+
+    /// Reverse the bits of the unsigned integer
+    ///
+    /// # Example
+    ///
+    ///```rust
+    /// use tfhe::integer::{gen_keys_radix, IntegerCiphertext};
+    /// use tfhe::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+    ///
+    /// let num_blocks = 4;
+    ///
+    /// // Generate the client key and the server key:
+    /// let (cks, sks) = gen_keys_radix(PARAM_MESSAGE_2_CARRY_2_KS_PBS, num_blocks);
+    ///
+    /// let msg = 0b10110100_u8;
+    ///
+    /// let ct = cks.encrypt(msg);
+    ///
+    /// // Compute homomorphically an addition:
+    /// let mut ct_res = sks.reverse_bits_parallelized(&ct);
+    ///
+    /// // Decrypt:
+    /// let res: u8 = cks.decrypt(&ct_res);
+    /// assert_eq!(msg.reverse_bits(), res);
+    /// ```
+    pub fn reverse_bits_parallelized<T>(&self, ct: &T) -> RadixCiphertext
+    where
+        T: IntegerRadixCiphertext,
+    {
+        let message_modulus = self.message_modulus().0 as u64;
+
+        let lut = self.key.generate_lookup_table(|x| {
+            (x % message_modulus).reverse_bits() >> (64 - message_modulus.ilog2())
```

Comment:
I think this function only works if carry bits have been cleaned, here as it's a default op it should do the cleaning see e.g. https://github.com/zama-ai/tfhe-rs/blob/91f05b00b9b0ff1fb3fb381f4a037c7997707395/tfhe/src/integer/server_key/radix_parallel/mul.rs#L579

for the classic bit of code to do that

Reply:
Done